build: don't allow Xcode as generator (#27715)
neovim/neovim: CMakeLists.txt

  cmake_policy(SET CMP0135 NEW)
endif()

if(XCODE)

>   message(FATAL_ERROR [[Xcode is not an allowed build generator at the moment. Use "Ninja" or "Unix Makefiles" instead.]])

Reviewer: ```suggestion
  message(FATAL_ERROR [[Xcode generator is not supported. Use "Ninja" or "Unix Makefiles" instead.]])
```
